libraries/src/Image/Image.php

```diff
@@ -116,12 +116,13 @@ public function __construct($source = null)
 		}
 
 		// Determine which image types are supported by GD, but only once.
-		if (!isset(static::$formats[IMAGETYPE_JPEG]))
+		if (empty(static::$formats))
 		{
 			$info = gd_info();
-			static::$formats[IMAGETYPE_JPEG] = ($info['JPEG Support']) ? true : false;
-			static::$formats[IMAGETYPE_PNG] = ($info['PNG Support']) ? true : false;
-			static::$formats[IMAGETYPE_GIF] = ($info['GIF Read Support']) ? true : false;
+			static::$formats[IMAGETYPE_JPEG] = $info['JPEG Support'];
+			static::$formats[IMAGETYPE_PNG]  = $info['PNG Support'];
+			static::$formats[IMAGETYPE_GIF]  = $info['GIF Read Support'];
+			static::$formats[IMAGETYPE_WEBP] = $info['WebP Support'];
```

Comment:
```suggestion
			static::$formats[IMAGETYPE_JPEG] = $info['JPEG Support'] ?? false;
			static::$formats[IMAGETYPE_PNG]  = $info['PNG Support'] ?? false;
			static::$formats[IMAGETYPE_GIF]  = $info['GIF Read Support'] ?? false;
			static::$formats[IMAGETYPE_WEBP] = $info['WebP Support'] ?? false;
```

I think we can't be sure that this parameters are always set.

Reply:
How come this wasn't an issue before?

Reply:
maybe because it's only a warning that maybe ignored in production systems. 

I don't know if you can always trust that this array parameters exists, it's more better safe then sorry but maybe unnecessary...

your decision.